src/Utils/ProductGalleryUtils.php

```diff
@@ -33,17 +33,21 @@ public static function get_product_gallery_images( $post_id, $size = 'full', $at
 				$size = $crop_images ? self::CROP_IMAGE_SIZE_NAME : $size;
 
 				foreach ( $all_product_gallery_image_ids as $product_gallery_image_id ) {
-					if ( $crop_images ) {
-						self::maybe_generate_intermediate_image( $product_gallery_image_id, self::CROP_IMAGE_SIZE_NAME );
+					if ( '0' !== $product_gallery_image_id ) {
```

Comment:
With this change, it would mean the cropping function won't work for the default placeholder image. I think we need to also account for the placeholder image because it is possible for store owners to use their own which may or may not be a square size.

Reply:
I think this is a bit of an edge case, at this point, so not high priority (IMO).
We'd also have to then implement a custom cropping function as we wouldn't be able rely on `maybe_generate_intermediate_image`.